assets/js/blocks/price-filter/block.js

```diff
@@ -5,13 +5,38 @@ import {
 	useQueryStateByKey,
 	useQueryStateByContext,
 	useCollectionData,
+	usePrevious,
 } from '@woocommerce/base-hooks';
 import { Fragment, useCallback, useState, useEffect } from '@wordpress/element';
 import PriceSlider from '@woocommerce/base-components/price-slider';
 import { CURRENCY } from '@woocommerce/settings';
 import { useDebouncedCallback } from 'use-debounce';
 import PropTypes from 'prop-types';
 
+const useConstraints = ( { minPrice, maxPrice } ) => {
+	const currentMinConstraint = isNaN( minPrice )
+		? null
+		: Math.floor( parseInt( minPrice, 10 ) / 10 ) * 10;
+	const previousMinConstraint = usePrevious( currentMinConstraint, ( val ) =>
+		Number.isFinite( val )
+	);
+	const minConstraint = Number.isFinite( currentMinConstraint )
+		? currentMinConstraint
+		: previousMinConstraint;
+
+	const currentMaxConstraint = isNaN( maxPrice )
+		? null
+		: Math.ceil( parseInt( maxPrice, 10 ) / 10 ) * 10;
+	const previousMaxConstraint = usePrevious( currentMaxConstraint, ( val ) =>
+		Number.isFinite( val )
+	);
+	const maxConstraint = Number.isFinite( currentMaxConstraint )
+		? currentMaxConstraint
+		: previousMaxConstraint;
+
+	return { minConstraint, maxConstraint };
+};
+
```

Comment:
Awesome idea to extract this to a hook.  As a side-effect, it also makes this behaviour more testable as it's isolated from the component.  Would you mind adding some tests for this hook that help verify it's behaviour?